src/protocol/context.rs

```diff
@@ -70,9 +71,7 @@ impl<'a, N, F> ProtocolContext<'a, N, F> {
             step: self.step.narrow(step),
             prss: self.prss,
             gateway: self.gateway,
-            // TODO: make this work
-            // accumulator: self.accumulator, //TODO: make this work
-            accumulator: None, // God help me, I just can't make this work
+            accumulator: self.accumulator.clone(),
```

Comment:
It was introducing that bound on `F` that helped me finally resolve this.

Reply:
Right.  You should now merge the two impls of ProtocolContext, as the second already had that bound.